src/decode_metar.c

```diff
@@ -1101,17 +1122,32 @@ static MDSP_BOOL isVisibility( char **visblty, Decoded_METAR *Mptr,
    char *achar,
         *astring,
         *save_token;
- 
- 
-   // CHECK FOR VISIBILITY MEASURED <1/4SM
- 
+   MDSP_BOOL isVisAfterM;
+
    if( *visblty == NULL )
-      return FALSE;
- 
- 
-   if( strcmp(*visblty,"M1/4SM") == 0 ||
-       strcmp(*visblty,"<1/4SM") == 0 ) {
-      Mptr->prevail_vsbySM = 0.0;
+       return FALSE;
+
+   /*
+    * Some visibility strings start with 'M' to indicate that they
+    * are actually some value less than the stated quantity, ie,
+    * < 1/4 SM is 'M1/4SM'.
+    */
+   if (*visblty[0] == 'M') {
+       visblty[0]++;
+       isVisAfterM = isVisibility(visblty, Mptr, NDEX);
+       visblty[0]--;
+       if (isVisAfterM) {
+           Mptr->visibilityIsUpperBound = TRUE;
+           return isVisAfterM;
+       }
+       return isVisAfterM;
+   }
+
+   // CHECK FOR VISIBILITY MEASURED <1/4SM
+
+   if( strcmp(*visblty,"<1/4SM") == 0 ) {
```

Comment:
After checking back over 6 months of metar history, the only visibility strings with "<" that I've seen are "<1/4SM" -- there aren't any `<\W?1/[1235789]SM` or `<\W?\d+SM`. Do you think it's worthwhile to handle "<" more generally in case data like that appears one day?

Reply:
If it's not too much additional code, it might be nice to handle both "<" and "M" cases generically.

Reply:
OK -- shouldn't be too hard to change